File: app/src/main/java/p/gordenyou/goproject/MainActivity.java
package p.gordenyou.goproject;

import android.os.Bundle;

import androidx.annotation.NonNull;

import p.gordenyou.common.ui.component.GoBaseActivity;
import p.gordenyou.goproject.logic.MainActivityLogic;


public class MainActivity extends GoBaseActivity implements MainActivityLogic.ActivityProvider {

    private MainActivityLogic activityLogic;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityLogic = new MainActivityLogic(this, savedInstanceState);
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        activityLogic.onSaveInstanceState(outState);
    }
}

File: app/src/main/java/p/gordenyou/goproject/MainActivity.kt
package p.gordenyou.goproject

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import p.gordenyou.goproject.demo.GoLogDemoActivity
import p.gordenyou.goproject.demo.tab.GoTabBottomDemoActivity
import p.gordenyou.goui.tab.bottom.GoTabBottom
import p.gordenyou.goui.tab.bottom.GoTabBottomInfo

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

    }

    fun onClick(view: View?) {
        when (view!!.id) {
            R.id.tv_hilog -> {
                startActivity(Intent(this, GoLogDemoActivity::class.java))
            }
            R.id.tv_bottom -> {
                startActivity(Intent(this, GoTabBottomDemoActivity::class.java))
            }
        }
    }
}